redash/query_runner/elasticsearch2.py

```
@@ -0,0 +1,292 @@
+
+        elif 'hits' in raw_result and 'hits' in raw_result['hits']:
+            for h in raw_result["hits"]["hits"]:
+                row = {}
+
+                fields_parameter_name = "_source" if "_source" in h else "fields"
+                for column in h[fields_parameter_name]:
+                    if result_fields and column not in result_fields_index:
+                        continue
+
+                    unested_results = get_flatten_results({column: h[fields_parameter_name][column]})
+
+                    for column_name, value in unested_results.items():
+                        add_column_if_needed(column_name, value=value)
+                        row[column_name] = value
+
+                result_rows.append(row)
+        else:
+            raise Exception("Redash failed to parse the results it got from Elasticsearch.")
+
+        return {
+            'columns': result_columns,
+            'rows': result_rows
+        }
+
+
+class OpenDistroSQLElasticSearch(ElasticSearch2):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.syntax = 'sql'
+
+    def _build_query(self, query: str) -> Tuple[dict, str, Optional[list]]:
+        sql_query = {
+            'query': query
+        }
+        sql_query_url = '/_opendistro/_sql'
+        return sql_query, sql_query_url, None
+
+
+class XPackSQLElasticSearch(ElasticSearch2):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.syntax = 'sql'
+
+    def _build_query(self, query: str) -> Tuple[dict, str, Optional[list]]:
+        sql_query = {
+            'query': query
+        }
+        sql_query_url = '/_sql'
+        return sql_query, sql_query_url, None
+
+    @classmethod
+    def _parse_results(cls, result_fields, raw_result):
+        error = raw_result.get('error')
+        if error:
+            raise Exception(error['reason'])
```

I ran into this branch and it eat the original exception: error was a string for me, which is not addressable by 'reason'.
Raising the error solved the problem: `raise Exception(error)`